Algorithmia/algorithm.py

```diff
@@ -29,6 +37,97 @@ def set_options(self, timeout=300, stdout=False, output=OutputType.default, **qu
         return self
 
+    # Create a new algorithm
+    def create(self, details={}, settings={}, version_info={}):
+        detailsObj = Details(**details)
+        settingsObj = SettingsMandatory(**settings)
+        createRequestVersionInfoObj = CreateRequestVersionInfo(**version_info)
+        create_parameters = {"name": self.algoname, "details": detailsObj, "settings": settingsObj, "version_info": createRequestVersionInfoObj}
+        create_request = CreateRequest(**create_parameters)
+        try:
+            # Create Algorithm
+            api_response = self.client.manageApi.create_algorithm(self.username, create_request)
+            return api_response
+        except ApiException as e:
+            error_message = json.loads(e.body)["error"]["message"]
+            raise ApiError(error_message)
+
+    # Update the settings in an algorithm
+    def update(self, details={}, settings={}, version_info={}):
+        detailsObj = Details(**details)
+        settingsObj = Settings(**settings)
+        createRequestVersionInfoObj = CreateRequestVersionInfo(**version_info)
+        update_parameters = {"details": detailsObj, "settings": settingsObj, "version_info": createRequestVersionInfoObj}
+        update_request = UpdateRequest(**update_parameters)
+        try:
+            # Update Algorithm
+            api_response = self.client.manageApi.update_algorithm(self.username, self.algoname, update_request)
+            return api_response
+        except ApiException as e:
+            error_message = json.loads(e.body)["error"]["message"]
+            raise ApiError(error_message)
+
+    # Publish an algorithm
+    def publish(self, details={}, settings={}, version_info={}):
+        detailsObj = Details(**details)
+        settingsObj = SettingsPublish(**settings)
+        versionRequestObj = VersionInfoPublish(**version_info)
+        publish_parameters = {"details": detailsObj, "settings": settingsObj, "version_info": versionRequestObj}
+        version_request = VersionRequest(**publish_parameters) # VersionRequest | Publish Version Request
+        try:
+            # Publish Algorithm
+            api_response = self.client.manageApi.publish_algorithm(self.username, self.algoname, version_request)
+            return api_response
+        except ApiException as e:
+            error_message = json.loads(e.body)["error"]["message"]
+            raise ApiError(error_message)
+
+    # Get info on an algorithm
+    def info(self, algo_hash=None):
+        try:
+            # Get Algorithm
+            if algo_hash:
+                api_response = self.client.manageApi.get_algorithm_hash_version(self.username, self.algoname, algo_hash)
+            else:
+                api_response = self.client.manageApi.get_algorithm(self.username, self.algoname)
+            return api_response
+        except ApiException as e:
+            error_message = json.loads(e.body)["error"]["message"]
+            raise ApiError(error_message)
```

Comment:
The way we're handling exceptions is throwing away information. What about the error code? stacktrace? etc? Probably doesn't belong in this PR, but something to think about.